libs/MeshKernel/include/MeshKernel/Utilities/RTreeSphericalToCartesian.hpp

```diff
@@ -0,0 +1,217 @@
+#include <MeshKernel/Entities.hpp>
+
+// include boost
+#define BOOST_ALLOW_DEPRECATED_HEADERS
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/point.hpp>
+#include <boost/geometry/index/rtree.hpp>
+#undef BOOST_ALLOW_DEPRECATED_HEADERS
+
+#include "MeshKernel/BoundingBox.hpp"
+#include "MeshKernel/Utilities/RTreeBase.hpp"
+
+#include <concepts>
+#include <utility>
+
+// r-tree
+// https://gist.github.com/logc/10272165
+
+namespace meshkernel
+{
+    // using a namespace alias
+    namespace bg = boost::geometry;
+    namespace bgi = boost::geometry::index;
+
+    /// @brief Class used for inquiring adjacent nodes in a mesh.
+    ///
+    /// The RTree class is designed for querying adjacent nodes within a mesh.
+    /// It encapsulates the boost::geometry::index::rtree functionality and extends it with a user-friendly
+    /// interface to perform common spatial queries. This class is templated on the projection type,
+    /// allowing flexibility in the geometric coordinate system used (default is bg::cs::cartesian).
+    ///
+    /// @tparam projection The geometric coordinate system projection (default is bg::cs::cartesian).
+    ///
+    /// The RTree class is primarily utilized within the mesh library for efficiently querying
+    /// the closest mesh nodes and edges to a specified point. It employs the RTreeBase class as its base.
+    ///
+    /// Internally, the RTree class maintains a query cache (`m_queryCache`) which is a vector used
+    /// to collect and store query results. This design helps optimize performance by avoiding frequent
+    /// reallocations when the number of results changes between queries.
+    ///
+    /// Example usage:
+    /// @code
+    /// // Create an RTree instance with the default Cartesian projection.
+    /// RTree<> cartesianRTree;
+    ///
+    /// // Perform a search for the nearest neighbors within a specified distance.
+    /// auto resultPoints = cartesianRTree.SearchPoints(queryPoint, searchDistance);
+    ///
+    /// // Perform a search for the single nearest neighbor.
+    /// auto nearestPoint = cartesianRTree.SearchNearestPoint(queryPoint);
+    /// @endcode
+    ///
+    /// Note: For advanced use cases and different geometric coordinate systems, users can provide
+    /// a custom projection template parameter when instantiating the RTree class.
+    ///
+    /// For more details on available query methods, refer to the base class documentation: meshkernel::RTreeBase.
+    class RTreeSphericalToCartesian : public RTreeBase
+    {
+        using Point2D = bg::model::point<double, 3, bg::cs::cartesian>; ///< Typedef for Point2D
+        using Box2D = bg::model::box<Point2D>;                          ///< Typedef for box of Point2D
+        using Value2D = std::pair<Point2D, UInt>;                       ///< Typedef of pair of Point2D and UInt
+        using RTree2D = bgi::rtree<Value2D, bgi::linear<16>>;           ///< Typedef for a 2D RTree
```

Comment:
This is something that I did not change, but probably should be changed is the name of these types.
Change from, e.g. Point2D to Point3D